build/search-index.js

```diff
@@ -18,7 +18,15 @@ module.exports = class SearchIndex {
 
   sort() {
     for (const items of Object.values(this._itemsByLocale)) {
-      items.sort((a, b) => getPopularity(b) - getPopularity(a));
+      items.sort((a, b) => {
+        const popularityA = getPopularity(a);
+        const popularityB = getPopularity(b);
+        const diff = popularityB - popularityA;
+        if (diff === 0) {
+          return a.url < b.url ? -1 : a.url > b.url ? 1 : 0;
+        }
+        return diff;
+      });
```

Comment:
I actually added this because I needed it for my benchmarking. 
I first wrote a bunch of implementations, some of which were async. Since we don't have a popularity for every possible URL, the final `searchIndex[$locale]` array became unpredictable. 

Before, when it was all sync the order of the became that of the order files were traversed. So for example: `/fr/docs/Web` came before `/fr/docs/Web/Foo`. Hopefully, I got the logic right this time but this new implementation yields the same results in the end.

Reply:
Looks like it should work 🤞